[docs] Document how to customize a subsection of a line chart (#13210)
mui/mui-x: docs/data/charts/line-demo/line-demo.md

## LineChartConnectNulls

{{"demo": "LineChartConnectNulls.js"}}

> ## Line with predicition

Reviewer: ```suggestion
## Line with multiple styles
```
I don't think this is specific for prediction, what do you think about having the "prediction" keyword in the example but not on the title?

Reviewer 2: The advantage of having it in the title is that it will be used by the Algolia search
So if we think people will search for this word, it's interesting
Not sure they will though

Reviewer: Algolia also searches for the body, I've added `estimated` and `styles` to allow for broader terms

Reviewer 2: > Algolia also searches for the body

With a similar importance?

Reviewer: No, but we also can't have a section title with all the keywords like a news blog 😂 `Line with multiple styles for prediction and estimations`

Author: since multiple and style are already used a lot, I prefer avoiding adding noise on those key words

ABout the discoverability of the example, I assume if it's hard to find we will get issues. It will be the opportunity to rethink how we organise the docs.

At the begining those pages were here mostly to compare our render with the Recharts examples (the reason why most of the demos are with weird title)

Reviewer: If we want to use market terms I would probably use `forecast` or `projection` instead of `prediction`? Though I'm not a specialist. Regardless, we at least have to fix the extra `i` in the current title.

Author: ```suggestion
## Line with forecast
```

> To show that some part of the data have a different status, you can vary the rendering.

Reviewer 2: What do you mean by "vary the rendering"?

Author: Triky to explain. The idea is to get multiple styles (in the example a line with dashed and non dashed area) on the same line

> To show that some part of the data have a different status, you can vary the rendering.

Reviewer: Maybe mention styling instead of rendering?

```suggestion
To show that parts of the data have different meanings, you can render stylised lines for each of them.
```

> In the following example, the chart shows some prediction.
> To do so, the `slots.line` is set with a custom components that render the default line twice.

Reviewer: ```suggestion
In the following example, the chart shows a dotted line to exemplify that the data is estimated.
To do so, the `slots.line` is set with a custom components that render the default line twice.
```
